### PR TITLE
[Merged by Bors] - TO-3305 actix with config

### PR DESCRIPTION
**Pull request by rustonaut**
Adds a multi layer configuration loading to the actix skeleton.

**References:**

- issue: [TO-3305]
- based on: #629
- blocked on: #630 (CI failure fixes)

### CODE REVIEW

**Comment by Robert-Steiner** [comment]
@rustonaut there are some merge conflicts, can I start reviewing or should i wait?

**Comment by rustonaut** [comment]
@Robert-Steiner you can review it

File: discovery_engine_core/web-api2/src/personalization.rs [hunk 1]
```diff
@@ -49,13 +47,17 @@ impl Application for Personalization {
...
+    common_config: CommonConfig,
 }
 
 impl Config for PersonalizationConfig {
     fn bind_address(&self) -> std::net::SocketAddr {
-        self.bind_to
+        self.common_config.bind_address()
+    }
+
+    fn log_file(&self) -> Option<&std::path::Path> {
```
[the remark this answers is not in the document]
Comment (rustonaut) [reply discovery_engine_core/web-api2/src/personalization.rs:59]: > would be even better if we had an option to disable writing to a file.

This is why it returns an option, it is not enabled as long as you don't explicitly enable it.

[the remark this answers is not in the document]
Comment (Robert-Steiner) [reply discovery_engine_core/web-api2/src/personalization.rs:59]: 👍 my bad! i didn't see the option

File: discovery_engine_core/web-api2/src/config.rs [hunk 2]
```diff
@@ -27,18 +30,87 @@ pub trait Config: DeserializeOwned + Send + Sync + 'static {
...
+///
+/// # Env and .env
+///
+/// Environment variables from `.env` and `.env.local` will be loaded in the process
+/// environment if they don't already exist there (keeping priority as described above).
+///
+/// For creating the config type instance only environment variables with the
+/// `XAYN_WEB_API_` prefix will be considered and the prefix is stripped.
+///
+/// Env variables are split by `__`. I.e. `XAYN_WEB_API_FOO__BAR=12` will be treated like
```
[the remark this answers is not in the document]
Comment (Robert-Steiner) [reply discovery_engine_core/web-api2/src/config.rs:85]: yeah should be `split`

File: discovery_engine_core/web-api2/src/config.rs [hunk 3]
```diff
@@ -27,18 +30,87 @@ pub trait Config: DeserializeOwned + Send + Sync + 'static {
...
+///
+/// # Env and .env
+///
+/// Environment variables from `.env` and `.env.local` will be loaded into the process
+/// environment if they don't already exist there (keeping priority as described above).
+///
+/// When creating the config type instance, only environment variables with the
+/// `XAYN_WEB_API_` prefix will be considered and the prefix is stripped.
+///
+/// Env variables are splitted at `__`. I.e. `XAYN_WEB_API_FOO__BAR=12` will be treated like
```
Comment (Robert-Steiner) [inline discovery_engine_core/web-api2/src/config.rs:85]: that was my bad 
```suggestion
/// Env variables are split at `__`. I.e. `XAYN_WEB_API_FOO__BAR=12` will be treated like
```

**Review by Robert-Steiner — APPROVED** [review]

**Comment by rustonaut** [comment]
bors merge

**Pull request closed without merging**